### ISSUE TITLE
TypeScript: Convert story provider

### ISSUE THREAD

**Issue by swissspidy** [swissspidy/web-stories-wp#12411]
## Task Description

This would be the start of a partial migration of the `story-editor` package.

The story provider is at the core of the editor and converting it to TS would be very helpful in making it more robust and of course type-safe.

**Comment by miina** [comment]
@swissspidy I started looking into this and noticed that in the `app/story` folder, the following non-typed packages are referenced:
- animation
- output
- design-system
- element-library

It also complains about `dom` even though that seems to be typed (not added to the `tsconfig.json` though).

Currently the issue is marked as blocked only by the `elements` package (which is actually typed already, pending a rework) but looks like there are more deps. A few options:
- Proceed and leave it as WIP waiting for the packages
- Ignore errors
- Continue and declare the missing modules locally with `any`
- Start looking at some of the deps instead for now

Thoughts?

**Comment by swissspidy** [comment]
Hmm I see.

In that case, local declarations and making any trivial conversions where possible would be my preference, just to unblock things so that we can move forward. But whatever works best for you.

> * animation
> * element-library

Since the code only references some constants, it should be possible to only convert `packages/animation/constants.js` and `packages/element-library/constant.js` to TS, with a simple tsconfig that only includes that file (like you did in #12619 basically)

> * design-system

Looks like only `useSnackbar` needs to be converted. Should be trivial to do that (in a separate PR perhaps)

> *output

Maybe for now just add very simple type declaration here locally.
